presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/HistogramStateSerializer.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.state;
+
+import com.facebook.presto.operator.aggregation.TypedHistogram;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.Type;
+
+import static com.facebook.presto.operator.aggregation.Histogram.EXPECTED_SIZE_FOR_HASHING;
+import static com.facebook.presto.spi.type.VarbinaryType.VARBINARY;
+
+public class HistogramStateSerializer
+        implements AccumulatorStateSerializer<HistogramState>
+{
+    private final Type type;
+
+    public HistogramStateSerializer(Type type)
+    {
+        this.type = type;
+    }
+
+    @Override
+    public Type getSerializedType()
+    {
+        return VARBINARY;
+    }
+
+    @Override
+    public void serialize(HistogramState state, BlockBuilder out)
+    {
+        if (state.get() == null) {
+            out.appendNull();
+        }
+        else {
+            VARBINARY.writeSlice(out, state.get().serialize());
```

Comment:
`serializedType` instead of `VARBINARY` this is more correct and will make it easier to integrate with @haozhun 's changes